src/orion/algo/bohb.py

```diff
@@ -0,0 +1,333 @@
+        self.eta = state_dict["eta"]
+        self.min_budget = state_dict["min_budget"]
+        self.max_budget = state_dict["max_budget"]
+        self.iteration = state_dict["iteration"]
+        self.iterations = state_dict["iterations"]
+        self.trial_meta = state_dict["trial_meta"]
+        self.trial_results = state_dict["trial_results"]
+        self.bohb = state_dict["bohb"]  # pylint: disable=attribute-defined-outside-init
+        self.strategy.set_state(state_dict["strategy"])
+        self._setup()
+
+    def suggest(self, num):
+        """Suggest a `num`ber of new sets of parameters.
+
+        TODO: document how suggest work for this algo
+
+        Parameters
+        ----------
+        num: int, optional
+            Number of trials to suggest. The algorithm may return less than the number of trials
+            requested.
+
+        Returns
+        -------
+        list of trials or None
+            A list of trials representing values suggested by the algorithm. The algorithm may opt
+            out if it cannot make a good suggestion at the moment (it may be waiting for other
+            trials to complete), in which case it will return None.
+
+
+        Notes
+        -----
+        New parameters must be compliant with the problem's domain `orion.algo.space.Space`.
+
+        """
+
+        def run_to_trial(run):
+            params = transform(run[1])
+            params[self.fidelity_index] = run[2]
+            return dict_to_trial(params, self.space)
+
+        def sample_iteration(iteration, trials):
+            while len(trials) < num and not iteration.is_finished:
+                run = iteration.get_next_run()
+                if run is None:
+                    break
+                new_trial = self.format_trial(run_to_trial(run))
+
+                # This means the job was already suggested and we have a result
+                result = self.trial_results.get(self.get_id(new_trial), None)
+                if result is not None:
+                    job = FakeJob(run, new_trial)
+                    job.result["loss"] = result
+                    iteration.register_result(job)
+                    self.bohb.new_result(job)
+                    continue
+
+                self.trial_meta.setdefault(self.get_id(new_trial), []).append(run)
+                self.register(new_trial)
+                trials.append(new_trial)
+
+        trials = []
+        for it in self.iterations:
+            sample_iteration(it, trials)
+        # If we don't have enough trials and there are still
+        # some iterations left
+        if self.iteration < len(self.budgets):
+            self.iterations.append(self._make_iteration(self.iteration))
+            self.iteration += 1
+            sample_iteration(self.iterations[-1], trials)
+
+        return trials
+
+    def observe(self, trials):
+        """Observe the `trials` new state of result.
+
+        TODO: document how observe work for this algo
```

Comment:
remove

Reply:
Fixed in d7ec60b22090108f9a6d70d41d9acd8df6dffa55